### PR TITLE
Progrium new

### PR DESCRIPTION
**Pull request by progrium**

### CODE REVIEW

File: pkg/misc/reflectutil/reflectutil.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package reflectutil
```
Comment (technoweenie) [inline pkg/misc/reflectutil/reflectutil.go:1]: I really like this API. :metal:

File: pkg/workspace/rpc/handlers.go [hunk 2]
```diff
@@ -350,7 +350,7 @@ func (s *Service) SetValue() func(qrpc.Responder, *qrpc.Call) {
 			return
 		}
 		n := s.State.Root.FindChild(params.Path)
-		//fmt.Println(n, params)
+		// fmt.Println(n, params)
```
Comment (technoweenie) [inline pkg/workspace/rpc/handlers.go:353]: Is this included as an intentional optional example to debug while developing?

File: pkg/workspace/ui/element.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package ui
+
+type Element struct {
+	Name     string
+	Attrs    Attrs
+	Children []Element
+}
+
+func E(name string, attrs Attrs, children []Element) Element {
+	return Element{name, attrs, children}
```
Comment (technoweenie) [inline pkg/workspace/ui/element.go:10]: Have you considered using a splat arg?

```go
func E(name string, attrs Attrs, children ... Element) Element {}

E("div", attrs, elem)
E("div", attrs, elem1, elem2)
E("div", attrs, elements...)
```

File: pkg/workspace/ui/state/state.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+			if ok {
+				customUI = uip.InspectorUI()
+			}
+
+			// look up the filepath for this component
+			var filepath string
+			if com.ID() != "" {
+				rc := library.LookupID(com.ID())
+				if rc == nil {
+					panic("component ID not in library: " + com.ID())
```
Comment (technoweenie) [inline pkg/workspace/ui/state/state.go:88]: What do you think about bubbling up an error from here to `InitializeDaemon()`? I can file an issue to investigate further.

File: studio/editors/ui/lib/atom.jsx [hunk 5]
```diff
@@ -5,16 +5,16 @@ export function Button(initial) {
...
-        view: function(vnode) {
+        view: function (vnode) {
             function press(e) {
                 state = "down";
             }
             function release(e) {
                 state = "up";
             }
-            return <div class={"Button rounded "+state+" "+vnode.attrs.class} style={{border: "2px solid black"}}>
+            return <div class={"Button rounded " + state + " " + vnode.attrs.class} style={{ border: "2px solid black" }}>
```
Comment (technoweenie) [inline studio/editors/ui/lib/atom.jsx:15]: Did you set up some kind of jsx linter? I'm in favor of these whitespace changes (or really, having _any_ project standard at all).

Comment (progrium) [reply studio/editors/ui/lib/atom.jsx:15]: yea i turned on the default "format on save" for vscode and this is what its been doing.

File: pkg/workspace/ui/state/state.go (hunk 4 above)
Comment (progrium) [reply pkg/workspace/ui/state/state.go:88]: yeah, is that possible? sounds like a good idea. this package is ripe for refactoring.

File: pkg/workspace/ui/element.go (hunk 3 above)
Comment (progrium) [reply pkg/workspace/ui/element.go:10]: yea that'd be good. will change it next time i use

File: pkg/workspace/rpc/handlers.go (hunk 2 above)
Comment (progrium) [reply pkg/workspace/rpc/handlers.go]: no just a commonly used debug print i uncomment sometimes that hasn't been refactored into logging system

File: pkg/misc/reflectutil/reflectutil.go (hunk 1 above)
Comment (progrium) [reply pkg/misc/reflectutil/reflectutil.go:1]: it wasn't a bad api before but i was trying to wrap reflect as methods and that wasn't working. starting to look a bit like this lib: https://github.com/tkrajina/go-reflector

i think this is the way to go, but i'd still like to see a value abstraction that gives you JS-like dynamic values. perhaps when/if necessary...

**Pull request merged**